index.src.html

```diff
@@ -2049,111 +2067,131 @@
+
+4. Let |resource| be a new [=frame resource=] allocated in accordance with
+    |frameInit|.
+
+    ISSUE: The spec should define explicit rules for each
+        {{PixelFormat}} and reference them in the steps above. See
+        [#165](https://github.com/w3c/webcodecs/issues/165).
+
+    NOTE: The user agent may choose to allocate resource with a larger coded
+      size and plane strides to improve memory alignment. Increases will be
+      reflected by {{VideoFrame/codedWidth}}, {{VideoFrame/codedHeight}}, and
+      {{Plane/stride}}.
+
+5. Let |resourceReference| be a reference to |resource|.
+6. Let |frame| be a new {{VideoFrame}} object initialized as follows:
+    1. Assign |resourceReference| to
+            {{VideoFrame/[[resource reference]]}}.
+    2. Assign |format| to {{VideoFrame/format}}.
+    3. Assign a new [=list=] to {{VideoFrame/planes}}.
+    4. For each |planeInit| in |planesInit|:
+        1. Copy |planeInit|.{{PlaneInit/src}} to |resource|.
+        2. Let |plane| be a new {{Plane}} initialized as follows:
+          1. Assign |frame| to {{Plane/[[parent frame]]}}.
+          2. Let |resourceStride| be the stride of the plane coresponding to
+              |planeInit| in |resource|.
+
+              ISSUE: The spec should provide a definition (and possibly
+                  diagrams) for stride. See
+                  [#166](https://github.com/w3c/webcodecs/issues/166).
+
+          3. Assign |resourceStride| to {{Plane/stride}}.
+          4. Assign |planeInit|.{{PlaneInit/rows}} to {{Plane/rows}}.
+          5. Assign the product of ({{Plane/stride}} * {{Plane/rows}}) to
+              {{Plane/length}}.
+        3. Append |plane| to {{VideoFrame/planes}}.
+
+    5. Let |resourceCodedWidth| be the coded width of |resource|.
+    6. Let |resourceCodedHeight| be the coded height of |resource|.
+
+        ISSUE: The spec should provide definitions (and possibly diagrams) for
+            coded size, crop size, and display size. See
+            [#166](https://github.com/w3c/webcodecs/issues/166).
+
+    7. Assign |resourceCodedWidth| to {{VideoFrame/codedWidth}}.
+    8. Assign |resourceCodedHeight| to {{VideoFrame/codedHeight}}.
+    9. Assign |init|.{{VideoFramePlanesInit/cropLeft}} to
+        {{VideoFrame/cropLeft}}.
+    10. Assign |init|.{{VideoFramePlanesInit/cropTop}} to
+        {{VideoFrame/cropTop}}.
+    11. Assign |init|.{{VideoFramePlanesInit/cropWidth}} to
+        {{VideoFrame/cropWidth}}.
+    12. Assign |init|.{{VideoFramePlanesInit/cropHeight}} to
+        {{VideoFrame/cropHeight}}.
+    13. Assign |init|.{{VideoFramePlanesInit/displayWidth}} to
+        {{VideoFrame/displayWidth}}.
+    14. Assign |init|.{{VideoFramePlanesInit/displayHeight}} to
+        {{VideoFrame/displayHeight}}.
+    15. Assign |init|.{{VideoFramePlanesInit/timestamp}} to
+        {{VideoFrame/timestamp}}.
+    16. Assign |init|.{{VideoFramePlanesInit/duration}} to
+        {{VideoFrame/duration}}.
```

Comment:
There an opportunity to clarify here, either by sharing steps or using English instead of repetitive steps.

Reply:
I condensed a few steps, but I also realized I was missing some defaulting behavior when things weren't provided. After fixing that, the opportunity to condense mostly disappeared (most steps have specific default logic that doesn't generalize).

High level: coded size is required (now by IDL), which cascades to crop size if not provided, and crop size cascades to display size if not provided.